attributes/attributes.go

```diff
@@ -121,13 +122,62 @@ func (a *Attributes) String() string {
 	return sb.String()
 }
 
-func str(x any) string {
-	if v, ok := x.(fmt.Stringer); ok {
-		return v.String()
+const nilAngleString = "<nil>"
+
+func str(x any) (s string) {
+	if x == nil { // NOTE: typed nils will not be caught by this check
+		return nilAngleString
+	} else if v, ok := x.(fmt.Stringer); ok {
+		return fmt.Sprint(v)
 	} else if v, ok := x.(string); ok {
 		return v
 	}
-	return fmt.Sprintf("<%p>", x)
+	value := reflect.ValueOf(x)
+	switch value.Kind() {
+	case reflect.Chan, reflect.Func, reflect.Map, reflect.Pointer, reflect.Slice, reflect.UnsafePointer:
+		return fmt.Sprintf("<%p>", x)
+	default:
+		// This will call badVerb to print as "<%p>", but without leading "%!(" and tailing ")"
+		return badVerb(x, value)
+	}
```

Comment:
Same here too. Let's get rid of the special handling in the `badVerb` function and live with what `%#v` gives us. Thank you.

Reply:
done.`ExampleAttributes_String` changed also.
+ before:
```go
	// Output:
	// a1: {"<attributes_test.key={}>": "<nil>" }
	// a2: {"<attributes_test.key={}>": "<nil>" }
	// a3: {"<attributes_test.key={}>": "<0x0>" }
	// a4: {"<attributes_test.key={}>": "<nil>" }
	// a5: {"<attributes_test.key={}>": "<int=1>" }
	// a6: {"<attributes_test.key={}>": "two" }
	// a7: {"<attributes_test.key={}>": "<attributes_test.stringVal={two}>" }
	// a8: {"<int=1>": "<bool=true>" }
```
+ after:
```go
	// Output:
	// a1: {"attributes_test.key{}": "<nil>" }
	// a2: {"attributes_test.key{}": "<nil>" }
	// a3: {"attributes_test.key{}": "(*attributes_test.stringVal)(nil)" }
	// a4: {"attributes_test.key{}": "<nil>" }
	// a5: {"attributes_test.key{}": "1" }
	// a6: {"attributes_test.key{}": "two" }
	// a7: {"attributes_test.key{}": "attributes_test.stringVal{s:\"two\"}" }
	// a8: {"1": "true" }
```